posts/serializers.py

```diff
@@ -1,6 +1,6 @@
 from rest_framework import serializers
 from .models import Post
-
+import uuid
 
 class PostSerializer(serializers.ModelSerializer):
     id = serializers.UUIDField()
```

Comment:
When you are using a foreign key and if you put that field in class Meta's fields list, please add that field's serializer here.
For example, you add author(a foreign key in line 25), then your code should be like 
```
from rest_framework import serializers
from .models import Post
import uuid
from users.serializers import UserSerializer


class PostSerializer(serializers.ModelSerializer):
    id = serializers.UUIDField()
    title = serializers.CharField(required=True)
    content = serializers.CharField(required=True)
    author = UserSerializer()

    def create(self, validated_data):
# more code...
```
If you don't do this, the returned json object will only contain the id of this foreign field. If you do this, it will return the nested json object. See difference
```
# if you add UserSerializer()
{
    "id": "b6839850-0ae0-494f-a0fd-3edc23b99d37",
    "title": "abc",
    "content": "test",
    "published": "2021-10-23T23:55:37.267595Z",
    "author": {
        "id": "8bc5847c-563d-48d9-a439-80923f6673d7",
        "user_name": "u1",
        "github_name": "abc"
    },
    "visibility": "PUBLIC"
}
```
```
# if you do not add UserSerializer()
{
    "id": "b6839850-0ae0-494f-a0fd-3edc23b99d37",
    "title": "abc",
    "content": "test",
    "published": "2021-10-23T23:55:37.267595Z",
    "author": "8bc5847c-563d-48d9-a439-80923f6673d7",
    "visibility": "PUBLIC"
}
```